rst2pdf/genpdftext.py

```diff
@@ -257,8 +257,13 @@ def get_pre_post(self, client, node, replaceEnt):
                 pre = u'<a name="%s"/>' % node['ids'][0]
                 client.targets.append(node['ids'][0])
         else:
-            pre = u'<a name="%s"/>' % node['refuri']
-            client.targets.append(node['refuri'])
+            for attr in ['refid', 'refuri']:
+                value = node.get(attr)
+                if value:
+                    pre = u'<a name="%s"/>' % value
+                    client.targets.append(value)
+                    break
+
```

Comment:
Do you want an else block to explicitly error out if neither of these are defined?

Reply:
Yeah, that would be good. Not sure what type of exception to raise though...

Reply:
I raised a ValueError. Hope that's a good choice.